obs_common/service_status.py

```diff
@@ -141,44 +100,40 @@ def print_delta(self, main_branch, user, repo, sha):
         self.row()
 
 
-def main():
-    config = get_config()
-
-    parser = argparse.ArgumentParser(description=DESCRIPTION)
+@click.command(help=DESCRIPTION)
+@click.option(
+    "--main-branch",
+    help=(
+        "The name of the main branch in the git repository. Defaults "
+        'to "main" if not configured in pyproject.toml.'
+    ),
+)
+@click.option(
+    "--host",
+    "hosts",
+    multiple=True,
+    help=(
+        "A list of hosts which have a ``/__version__`` Dockerflow endpoint in the "
+        "form of ENVIRONMENTNAME=HOST."
+    ),
+)
+def main(main_branch, hosts):
+    config_data = {}
+    if (pyproject_toml := Path("pyproject.toml")).exists():
+        data = tomllib.loads(pyproject_toml.read_text())
+        config_data = data.get("tool", {}).get("service-status", {})
+
+    if main_branch is None:
+        main_branch = config_data.get("main_branch", "main")
```

Comment:
I tend to use `or` for things like this especially where a falsey empty string doesn't make sense:

```suggestion
    main_branch = main_branch or config_data.get("main_branch", "main")
```

Reply:
I would agree, but in this case the falsey string would have to be explicitly passed as a cli flag, and I don't think we should ignore that

Reply:
i slept on it an changed me mind